src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -29,6 +32,66 @@ public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 			return builder;
 		}
 
+		public static MauiAppBuilder ConfigureEnvironmentVariables(this MauiAppBuilder builder)
```

Comment:
I'm not sure I understand why we need to do this. Why can't `Microsoft.Extensions.Configuration.EnvironmentVariables` be used instead?

Reply:
If we're running on iOS or Android the environment variables on those devices/Simulators/emulators are not necessarily the ones that are on the host machine.

In the case of Android we write them to a file through the Hot Reload connection and I read them directly from there as its more stable that way. For iOS we provide them through the debugger connection which works fine.

However, it doesn't end there, I want to add some magic for not using localhost since that is also something that won't work for Android.

Having all that said, I do think we might want to look into still implementing it. I remember us having some open issues to support the settings.json files etc. so it might still be beneficial.